main.cpp

```diff
@@ -128,421 +180,401 @@ struct CarWash
+   switchedOn = false;
+}
+
+
+
 
 struct Dentist 
 {
     
-    //number of patients per day
     int numPatients = 8;
-    //maximum number of patients per week
     int maxPatientsPerWeek = 40;
-    //disposable equipment costs per week
-    double costDisposableEquipment = 2000;
-    //profit per week (float)
-    double profitPerWeek = 20000;
-    //number of staff (int)
+    int costDisposableEquipment = 2000;
+    int profitPerWeek = 20000;
     int numStaff = 20;
 
     struct Person
     {
-        //personName
         std::string name = "Mary";
-        //person age in years
         int age = 35;
-        //person gender
         std::string gender = "female";
-        //personInsuranceCompany 
         std::string insuranceCompany = "No Insurance";
-        //date of last check (ddmmyyyy)
-        int dateOfLastCheck = 01112022;
-
-        //return patient's insurance company
-        std::string getInsuranceCompany(std::string insuranceCompany); //returns name of patient's insurance company
-        //dental treatment deemed necessary
-        int dateOfNextDentalAppointment(int dateOfLastCheck); //returns month when next check will be
-        //cost of work undertaken
-        float costOfDentalTreatment(); //returns cost of treatment
+        int dateOfLastCheck = 112022;
+
+        std::string getInsuranceCompany(std::string insurance); 
+        int dateOfNextDentalAppointment(int lastCheck); 
+        void setGender(std::string gen); 
     };
 
-    //treat patient
-    void treatPatient(Person name, std::string treatmentClassification);
-    //charge patient fee for service
-    float feeForService(std::string treatmentClassification); //returns cost in dollars of treatment
-    //pay staff wages
-    float payStaff(std::string name); //returns staff member's salary
+    int returnLastCheck(std::string name);
+    float feeForService(std::string treatmentClassification);
+    int payStaff(); 
 
     Person newPatient;
 };
 
-/*
-Thing 3) Automatic public toilet
-5 properties:
-    1) volume of cistern (float)
-    2) volume of cleaning solution reservoir (float)
-    3) number of uses per day (int)
-    4) brightness of lights (int)
-    5) amount of water per flush (float)
-3 things it can do:
-    1) flush
-    2) illuminate
-    3) clean
- */
+
+std::string Dentist::Person::getInsuranceCompany(std::string insurance)
+{
+    insuranceCompany = insurance;
+    return insuranceCompany;
+}
+
+int Dentist::Person::dateOfNextDentalAppointment(int lastCheck)
+{
+    dateOfLastCheck = lastCheck;
+    return dateOfLastCheck + 1;
+}
+
+void Dentist::Person::setGender(std::string gen)
+{
+    gender = gen;
+}
+
+int Dentist::returnLastCheck(std::string name)
+{
+    newPatient.name = name;
+    return newPatient.dateOfLastCheck;
+}
+
+float Dentist::feeForService(std::string treatmentClassification)
+{
+    if( treatmentClassification == "check up" )
+    {
+        return 500;
+    }
+    else
+    {
+        return 1000;
+    }
+}
```

Comment:
If you find yourself writing code like this:
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    else 
    {
        return anotherValue;
    }
}
```
You do not need the `else` block.  Here's why:
`return` statements are the last thing that **ever happens** in a function.
When the `if()` expression is evaluated as `true` and the block is entered, the `return aValue;` is the last line of code to be executed before the function exits.
but when the `if()` expression evaluates as `false` and the block is **NOT** entered, the next thing that happens is `return anotherValue;`
you have no code after the `else` block that needs executing, since your `else` block contains a `return` statement.
that means you do **NOT** need the `else` block, and it is just causing visual clutter.
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    
    return anotherValue;
}
```
- [ ] revise

Reply:
Thanks -- I should have spotted this as its in the lecture.